src/ConcordNet/ConcordGenerator.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.IO;
+using System.Text.Json;
+using ConcordNet.Interfaces;
+
+namespace ConcordNet
+{
+    public class ConcordGeneratorConfig
+    {
+        public string ContractDirectory { get; set; }
+    }
+    public class ConcordGenerator : IConcordGenerator
+    {
+        private string contractDirectory;
+        
+        private string Consumer;
+        private string Provider;
+        
+        private IMockProviderService _providerService;
+
+
+        public ConcordGenerator(ConcordGeneratorConfig config)
+        {
+            contractDirectory = config.ContractDirectory;
+            if (!contractDirectory.EndsWith("/"))
+            {
+                contractDirectory += "/";
+            }
+
+            if (!Directory.Exists(contractDirectory))
+            {
+                Directory.CreateDirectory(contractDirectory);
+            }
+        }
+        public IMockProviderService MockService(int port)
+        {
+            _providerService = new MockProviderService(port);
+            return _providerService;
+        }
+
+        public IConcordGenerator ServiceConsumer(string ServiceConsumer)
+        {
+            Consumer = ServiceConsumer;
+            return this;
+        }
+
+        public IConcordGenerator HasContractWith(string providerService)
+        {
+            Provider = providerService;
+            return this;
+        }
+
+        [Obsolete]
```

Comment:
What is the reason for this being obsolete? Could you put a reason on this?

Reply:
Ohh sure,  this was so we could do minimal changes in the APIs but would get compiler warnings

Reply:
Honestly I'll just remove it